### ISSUE TITLE
Clarification of Link-Faux 

### ISSUE THREAD

**Issue by steveduffin** [sky-uk/toolkit#286]
Currently we have `c-link-faux` in toolkit, but their are different use cases for making a button act like a link and each require different styling.

1. c-link-faux currently addresses the need to have a button that is presented as text as part of a control - often aligned with or consisting of an icon. The focus is styled with an offset shadow.

![screen shot 2017-09-28 at 13 12 40](https://user-images.githubusercontent.com/1849493/30966492-000e28ae-a451-11e7-9660-650248e1fe5c.png)

![screen shot 2017-09-28 at 13 25 09](https://user-images.githubusercontent.com/1849493/30966498-040e046a-a451-11e7-8914-4cc76ce6de58.png)

Suggestion: make this a modifier of `.c-btn` e.g `.c-btn--naked`

2. There is also the need for a link to become a button, purely for semantic reasons but should be presented as standard `<a>` i.e use the brand colour (blue), focus & hover with an underline.

![screen shot 2017-09-28 at 13 08 25](https://user-images.githubusercontent.com/1849493/30966524-1c8fc2e4-a451-11e7-8b0a-b648323b789f.png)

![screen shot 2017-09-28 at 13 30 36](https://user-images.githubusercontent.com/1849493/30966562-3f9ae462-a451-11e7-9888-44579da465ce.png)

![screen shot 2017-09-28 at 13 29 37](https://user-images.githubusercontent.com/1849493/30967023-ca98047c-a452-11e7-99df-bb37ec2ffdba.png)

^ Examples above use `<button>` as there is no href and trigger an internal action on the page.

This has been implemented already by Sky Mobile as `.u-faux-link`, is part of a major piece of upcoming work *(NAME REDACTED)* and needs addressing on the Terms and Conditions toolkit-react competent used in Loyalty and *(NAME REDACTED)*.

**Comment by MrDinsdale** [comment]
My only real concern is that the only similarity `c-btn` and `c-btn--naked` share is that they are intended to be used on a button element (although `c-btn` can be used on `<a>`). I'm not sure its worth overwriting all the default `c-btn` styles to reset its appearance.

I would argue `c-link-faux` is still the neatest approach although we need to add a modifier for the `color(brand)` blue to bring inline with other links and some minor updates to the docs.

**Comment by steveduffin** [comment]
Inclined to agree about the button modifier.

If we want to go in a different direction with the focus style with this new faux-link, then do we want to be resting the pseudo-element that `c-link-faux` would add?

**Comment by joebell93** [comment]
So one key thing here that hasn't been mentioned is **box-model**

Buttons have padding.
Link-Faux does **not**.

This is so that link-faux sits on the container edge for perfect alignment in cases such as a panel close button, whilst providing the same padding via a pseudo element to provide the same focus styles

Doesn't feel right to completely change `c-btn`'s structure via a button modifier, a separate component (i.e. link-faux) feels more appropriate in that case

**Comment by joebell93** [comment]
Also regarding colours, I'd like to get @mikejgregory's thoughts on this. I'll have a chat tomorrow and keep you posted

**Comment by steveduffin** [comment]
To clarify a few points: 

I'm suggesting we keep `link-faux` as is but give it an additional name that in toolkit 3.0, replaces it. For use case 1, it's doing a good job, alignment and all. 
However, calling it `link-faux` is confusing AF as it's not styled as `a:link` and it's not a `btn`. It's it's own thing. Does it always involve an icon? Naming suggestions welcome. 

I agree that for use case 2, our new style should **not** be a button modifier. But I don't think it's helpful to be a modifier of `link-faux` either. If there wasn't a `link-faux` already, I'd want to call it that. But now we need to distinguish it from use case 1, so a new name is needed. `.c-link` and add it to `links.scss`?

**Comment by joebell93** [comment]
I'm not sure, adding `c-link` to a `btn` doesn't really make sense to me - I think we're better off getting one decision from design on how it should look before we jump to a solution

**Comment by mikejgregory** [comment]
This is still confusing for design (well me).  From a design stand point we should have a button suite which consists of; primary, secondary, select, invert, icon and loading.  This should be expanded out to cover icon buttons without a container so it covers "close" and "back" etc for panels.  Then on top of this we have text links which should always be styled in the same way.  Blue text which underlines on hover.  For design there is a clear difference between buttons and links. I know dev won't class some of the above as a button but within design we do.

**Comment by mikejgregory** [comment]
Happy to sit down and sort this out together as I suspect this won't be solved on here.

**Comment by steveduffin** [comment]
I think/hope we are all on the same page but it's now a question of semantics. Let's bash this one out in the next meetup

**Comment by joebell93** [comment]
Just a thought after the Slack discussion, maybe we need a "text/naked" button" but one that can be blue or black? (like how the costing modifiers can be chained)

```
.c-btn--naked.c-btn--brand
```

Just thinking aloud, not even sure I like that. I'm just not too sure on `.c-link`.

It essentially seems to boil down to the colour right? I think we could still use with some design steering on this, why would we have blue vs. black? Is it purely cosmetic or is there some logic behind this?

**Comment by steveduffin** [comment]
It's not just the colour. It needs to strip everything off a browser's treatment of a `<button>` and dress it up with hover and focus effects. It has no connection to a `c-btn`

**Comment by joebell93** [comment]
Aye, that's all the features `.c-link-faux` provides

**Comment by steveduffin** [comment]
`.c-link-faux` only does a partial reset of button styles, then adds a bunch of custom stuff for it's focus and hover that is only useful in the use case 1 (with a close icon for example).

e.g it's not resetting the `appearance: button, it adds `vertical-align: middle; ` and adds a pseudo-element to build a custom focus.

**Comment by steveduffin** [comment]
Codepen showing current issue with using `c-link-faux` for use case 2 and proposed solution https://codepen.io/steveduffin/pen/LzdJpj

**Comment by joebell93** [comment]
Discussion notes:

Short term, we propose a new modifier that overrides `link-faux` for it's intended behaviour in Toolkit 3.0 (this means that updating to 3.0 when using this class, won't break your intended design)

Long term, we propose adding a "text button"-style (naming tbc) component that lives within `components/buttons` to provide the black text underline button - to be raised in a separate issue
